addons/mail/static/src/core/common/chat_window_service.js

```diff
@@ -40,7 +40,7 @@ export class ChatWindowService {
             return rpc(
                 "/discuss/channel/fold",
                 {
-                    channel_id: chatWindow.thread.id,
+                    channel_id: parseInt(chatWindow.thread.id),
```

Comment:
Apparently, `chatWindow.thread.id` is a proxy and is converted to an array when doing the RPC call. As the route `/discuss/channel/fold` and the other functions expect a single integer, I will cast the proxy into an integer.

Reply:
chatWindow/thread might be proxy but it seems impossible for `id` to be a proxy, unless it doesn't contain a number in which case that is the root cause to fix

Reply:
I investigate a bit and I confirm that `chatWindow.thread.id` is a proxy in the test environment. It seems like: When I call the service to open a chat window, the service returns an object `chatWindow` that is created by the `preinsert` method of `@mail/core/common/record.js`. The issue should be hidden somewhere in there but I don't understand much of that code.

As it is really technical and my knowledge of Discuss is limited and my browser crashes when I add a breakpoint in the JS test suite :clown_face:, may I come to your office to check this out with you?

Reply:
I'll try locally first, what is the easiest to have the issue if I remove the cast to int?

Reply:
I just ran my own test `Can open lead from internal link` in debug mode.

Reply:
I checked and as I suspected `id` does not contain a number but an array.

It comes from `originThread: { model: "discuss.channel", id: args.args[0] }` where `args.args[0]` is ids (array).

I'll push the correct version so I can approve the PR immediately.